### ISSUE TITLE
"combine_lists_err: different length lists" error

### ISSUE THREAD

**Issue by vzaliva** [vellvm/vellvm#103]
Interpreting attached code gives the following error:

```
$ ./vellvm -interpret ~/src/helix/tests/test_binop_plus.ll       15:08:32
(* -------- Vellvm Test Harness -------- *)
Processing: /home/lord/src/helix/tests/test_binop_plus.ll
Fatal error: exception Failure("combine_lists_err: different length lists")
```

[test_binop_plus.ll](https://github.com/vellvm/vellvm/files/3721949/test_binop_plus.ll.txt)

**Comment by YaZko** [comment]
Hello Vadim,

Thanks for the report. Indeed we do not yet have a story for main arguments, and went for a quick hack to handle our test suite by providing default values to argc and argv. Unfortunately, it ended up requiring them in the prototype as a side-effect.

I pushed a quick fix. it's unsanitary (see Issue #104) but should let you run code with argument-less main functions.

Your file also exhibited a critical bug in our handling of phi nodes. It should be fixed by 3241f11b4089c86094fed00d0d64f3e134e03796

With these two patches, your file runs without failure, and appears to agree with llc.

Best,
Yannick

**Comment by vzaliva** [comment]
Thanks. That seems to help with this error messages but exposed some others :)
I will file separate reports.
